### ISSUE TITLE
Error Running Colcon Build

### ISSUE THREAD

**Issue by keenanjohnson** [ros2-rust/ros2_rust#85]
Hello! I was trying to install the ros2_rust examples in a docker container according to the instructions and I am currently running into an error with the colcon build command.

When I run colcon build exactly as described, I get the following error `colcon: error: unrecognized arguments: --packages-up-to`

Here's my dockerfile that I'm trying to use for reference. Any help would be appreciated! I think I followed the instructions correctly, but I could be missing something obvious. Thank you!

```
FROM balenalib/%%BALENA_ARCH%%-ubuntu:focal as base
ARG DEBIAN_FRONTEND=noninteractive

##########################
## ROS2 Foxy Install
##########################

# Install language support for ROS
RUN apt-get update && apt-get install -y \
  locales \
  && locale-gen en_US.UTF-8 \
  && update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 \
  && rm -rf /var/lib/apt/lists/*
ENV LANG en_US.UTF-8

# Install ROS2
RUN apt-get update && apt-get install -y \
    curl \
    gnupg2 \
    lsb-release \
    sudo \
  && curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg \
  && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null \
  && apt-get update && apt-get install -y \
    ros-foxy-ros-base \
    ros-foxy-test-interface-files \
    python3-argcomplete \
  && rm -rf /var/lib/apt/lists/*

# Configure Enviroment for ROS
RUN echo "source /opt/ros/foxy/setup.bash" >> ~/.bashrc

##########################
## Rust Install
##########################
ENV PATH=/root/.cargo/bin:$PATH
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain 1.59.0 -y

##########################
## ros2_rust install
##########################
RUN apt-get update && apt-get install -y \
    python3-vcstool \
    libclang-dev \
    clang \
    python3.9 \
    python3.9-dev \
    python3-pip \
    git \
    && rm -rf /var/lib/apt/lists/*

# Install the colcon-cargo and colcon-ros-cargo plugins
RUN pip install git+https://github.com/colcon/colcon-cargo.git git+https://github.com/colcon/colcon-ros-cargo.git
RUN pip install --upgrade pytest
# Install the cargo-ament-build plugin
RUN cargo install cargo-ament-build

RUN mkdir src
RUN git clone https://github.com/ros2-rust/ros2_rust.git src/ros2_rust
RUN vcs import src < src/ros2_rust/ros2_rust_foxy.repos
RUN . /opt/ros/foxy/setup.sh
RUN colcon build --packages-up-to rclrs_examples

CMD sleep infinity
```

**Comment by keenanjohnson** [comment]
Chatting with @nnmm about this, we've made some progress. Documenting here for the record.

Two issues discover with my process above:

1. I was using the Base version of Ros instead of Desktop, which seemed to be missing some important things like colcon-package-selection. I resolved this by switching to ROS Desktop. I also just decided to switch to the official ROS2 Foxy Docker image.
2. The `colcon build --packages-up-to rclrs_examples` command was being run in the `/` dir instead of the `/src/` dir.

I resolved both of these issues and created this Dockerfile:

```
# Ros2 Foxy runs on ubuntu focal
FROM ros:foxy as base
ARG DEBIAN_FRONTEND=noninteractive

##########################
## Rust Install
##########################
# Install Dependancies
RUN apt-get update && apt-get install -y \
    curl \
    && rm -rf /var/lib/apt/lists/*

RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain 1.59.0 -y
ENV PATH=/root/.cargo/bin:$PATH

##########################
## ros2_rust install
##########################
# Install Dependancies
RUN apt-get update && apt-get install -y \
    python3-vcstool \
    libclang-dev \
    clang \
    python3.9 \
    python3.9-dev \
    python3-pip \
    git \
    && rm -rf /var/lib/apt/lists/*

RUN pip install --upgrade pytest colcon-package-selection

# Install the colcon-cargo and colcon-ros-cargo plugins
RUN pip install git+https://github.com/colcon/colcon-cargo.git git+https://github.com/colcon/colcon-ros-cargo.git
# Install the cargo-ament-build plugin
RUN cargo install cargo-ament-build

# Install the ros2_rust package
RUN mkdir src
RUN git clone https://github.com/ros2-rust/ros2_rust.git src/ros2_rust
RUN vcs import src < src/ros
```2_rust/ros2_rust_foxy.repos
RUN . /opt/ros/foxy/setup.sh
WORKDIR /src/
RUN colcon build --packages-up-to rclrs_examples

##########################
## Build Rust App
##########################
WORKDIR /usr/src/
COPY . .

RUN cargo install --path .

CMD ["flight-controller"]
```

Building this image results in the following error:

```
[main]     Step 14/18 : RUN colcon build --packages-up-to rclrs_examples
[main]      ---> Running in 4e6353cdfe09
[main]     Starting >>> rosidl_runtime_rs
[main]     --- stderr: rosidl_runtime_rs
[main]     Traceback (most recent call last):
[main]       File "/usr/lib/python3/dist-packages/colcon_core/executor/__init__.py", line 91, in __call__
[main]         rc = await self.task(*args, **kwargs)
[main]       File "/usr/lib/python3/dist-packages/colcon_core/task/__init__.py", line 93, in __call__
[main]         return await task_method(*args, **kwargs)
[main]       File "/usr/local/lib/python3.8/dist-packages/colcon_ros_cargo/task/ament_cargo/[build.py](http://build.py/)", line 68, in build
[main]         return await super(AmentCargoBuildTask, self).build(
[main]       File "/usr/local/lib/python3.8/dist-packages/colcon_cargo/task/cargo/[build.py](http://build.py/)", line 49, in build
[main]         rc = await self._build(args, env)
[main]       File "/usr/local/lib/python3.8/dist-packages/colcon_ros_cargo/task/ament_cargo/[build.py](http://build.py/)", line 84, in _build
[main]         new_package_paths = find_installed_cargo_packages(env)
[main]       File "/usr/local/lib/python3.8/dist-packages/colcon_ros_cargo/task/ament_cargo/[build.py](http://build.py/)", line 142, in find_installed_cargo_packages
[main]         for prefix in env['AMENT_PREFIX_PATH'].split(os.pathsep):
[main]     KeyError: 'AMENT_PREFIX_PATH'
[main]     ---
[main]     Failed   <<< rosidl_runtime_rs [0.02s, exited with code 1]
[main]     
[main]     Summary: 0 packages finished [0.51s]
[main]       1 package failed: rosidl_runtime_rs
[main]       1 package had stderr output: rosidl_runtime_rs
[main]       9 packages not processed
[main]     'AMENT_PREFIX_PATH'
[main]     
[main]     Removing intermediate container 4e6353cdfe09
[main]     The command '/bin/sh -c colcon build --packages-up-to rclrs_examples' returned a non-zero code: 1
```

**Comment by keenanjohnson** [comment]
Had to combine the ROS overlay command with the colcon build command to satisfy Docker, but that has resulted in a new error:

```
[main]     Starting >>> rclrs_examples
[main]     --- stderr: rclrs_examples
[main]     thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-ament-build-0.1.2/src/main.rs:33:62
[main]     note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[main]     ---
[main]     Failed   <<< rclrs_examples [0.21s, exited with code 101]
[main]     
[main]     Summary: 9 packages finished [1min 28s]
[main]       1 package failed: rclrs_examples
[main]       2 packages had stderr output: rclrs rclrs_examples
[main]     Removing intermediate container 69cf8b900e43
[main]     The command '/bin/sh -c . /opt/ros/foxy/setup.sh && colcon build --packages-up-to rclrs_examples' returned a non-zero code: 101
```

Complete Dockerfile

```
# Ros2 Foxy runs on ubuntu focal
FROM ros:foxy as base
ARG DEBIAN_FRONTEND=noninteractive

##########################
## Rust Install
##########################
# Install Dependancies
RUN apt-get update && apt-get install -y \
    curl \
    && rm -rf /var/lib/apt/lists/*

RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain 1.59.0 -y
ENV PATH=/root/.cargo/bin:$PATH

##########################
## ros2_rust install
##########################
# Install Dependancies
RUN apt-get update && apt-get install -y \
    python3-vcstool \
    libclang-dev \
    clang \
    python3.9 \
    python3.9-dev \
    python3-pip \
    git \
    && rm -rf /var/lib/apt/lists/*

RUN pip install --upgrade pytest colcon-package-selection

# Install the colcon-cargo and colcon-ros-cargo plugins
RUN pip install git+https://github.com/colcon/colcon-cargo.git git+https://github.com/colcon/colcon-ros-cargo.git
# Install the cargo-ament-build plugin
RUN cargo install cargo-ament-build

# Install the ros2_rust package
RUN mkdir src
RUN git clone https://github.com/ros2-rust/ros2_rust.git src/ros2_rust
RUN vcs import src < src/ros2_rust/ros2_rust_foxy.repos

# Build rclrs
WORKDIR /src/
RUN . /opt/ros/foxy/setup.sh && colcon build --packages-up-to rclrs_examples

##########################
## Build Rust App
##########################
WORKDIR /usr/src/
COPY . .

RUN cargo install --path .

CMD ["flight-controller"]

```

**Comment by keenanjohnson** [comment]
This error was related to a bug in cargo-ament-build 0.1.2. 

@nnmm pushed a fix in 0.1.3 https://crates.io/crates/cargo-ament-build/0.1.3 which has allowed me to compile with the dockerfile above!
